File: ExtractResourceToFile/Extensions/ResourceExtensions.vb
Imports System.IO

Namespace Extensions

    Module ResourceExtensions
        <Runtime.CompilerServices.Extension()>
        Public Sub FileSave(BytesToWrite() As Byte, FileName As String)

            If File.Exists(FileName) Then
                File.Delete(FileName)
            End If

            Dim FileStream As New FileStream(FileName, FileMode.OpenOrCreate)
            Dim BinaryWriter As New BinaryWriter(FileStream)

            BinaryWriter.Write(BytesToWrite)
            BinaryWriter.Close()
            FileStream.Close()

        End Sub
    End Module
End Namespace

File: ExtractResourceToFile1/Extensions/ResourceExtensions.cs
using System.IO;

namespace ExtractResourceToFile1.Extensions
{
    public static class ResourceExtensions
    {
        /// <summary>
        /// Given a project resource extract to desired location
        /// </summary>
        /// <param name="BytesToWrite">Resource</param>
        /// <param name="FileName">File name with extension and path is optional</param>
        public static void FileSave(this byte[] BytesToWrite, string FileName)
        {

            if (File.Exists(FileName))
            {
                File.Delete(FileName);
            }

            var FileStream = new FileStream(FileName, FileMode.OpenOrCreate);
            var BinaryWriter = new BinaryWriter(FileStream);

            BinaryWriter.Write(BytesToWrite);
            BinaryWriter.Close();
            FileStream.Close();

        }
    }
}
